samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+                connection.setAutoCommit(true);
+            }
+
+            PreparedStatement statement = connection.prepareStatement(config.getQuery());
+            int batchCounter = config.getBatchSize();
+            int batchRound = 0;
+
+            for (int i = 0; i < config.getIterations(); i++) {
+                if (config.getBatchSize() == 0) {
+                    // single statements
+                    try {
+                        if (config.getQueryParams() != null) {
+                            statement = parametrizeStatement(statement, config.getQueryParams());
+                        }
+
+                        if (isGoogleSQL) {
+                            connection.createStatement().execute(String.format("SET STATEMENT_TAG='%s'",
+                                    DEFAULT_TAG));
+                        } else {
+                            connection.createStatement().execute(String.format("SET SPANNER.STATEMENT_TAG='%s'",
+                                    DEFAULT_TAG));
+                        }
+
+                        boolean hasResults = false;
+                        long start = 0;
+                        long stop = 0;
+                        if (connection.getAutoCommit()) {
+                            // read-only
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            stop = System.nanoTime() - start;
+                        } else {
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            connection.commit();
+                            stop = System.nanoTime() - start;
+                        }
+
+                        if (hasResults) {
+                            statement.getResultSet().close();
+                        }
+
+                        measures[i] = stop / 1000000;
+                        progress++;
+                    } catch (Exception e) {
+                        if (e.getMessage().contains("ALREADY_EXISTS:")) {
+                            System.out.println("duplicate key - retry");
+                            i--;
+                        } else {
+                            throw e;
+                        }
+                    }
+                } else if (config.getQuery().contains("INSERT") ||
+                        config.getQuery().contains("UPDATE")
+                        || config.getQuery().contains("DELETE")) {
+                    // batching
+                    try {
+                        if (config.getQueryParams() != null) {
+                            statement = parametrizeStatement(statement, config.getQueryParams());
+                        }
+
+                        statement.addBatch();
+
+                        if (batchCounter == 0 || i == config.getIterations() - 1) {
```

Comment:
This could use a comment. I think I understand how it works (`batchCounter` is decremented for every loop, and then reset to `batchSize` after executing a batch)